### PR TITLE
explain how to get system dependencies

### LINKED ISSUE #3: Instructions for cross-compiling ffi packages
Hi,
These instructions worked nicely, thanks - I'm impressed how things just work!
However, it doesn't cover using packages with foreign dependencies; in this case I was trying to use one of the sqlite3 crates. I'd be interested in some hints for making those work if you have any! 

Thanks again for the very useful guide!

### PR DESCRIPTION
**Pull request by tfnico**
In case people use libraries that have system dependencies, they have to extend the toolset
with these. Show some examples on how to do this.

Closes #3.

### CODE REVIEW

**Comment by Ogeon** [comment]
Thank you so much! I will take a closer look as soon as I get back home. Until then, it would be nice if you can add a reference to this section from the part where the sysroot is set up. Could you also rename it to something like "Appendix A: ..."? It would make it more consistent if more appendixes would be added later.

Also, I think this would close the issue, so you can edit the description to say "Closes #3".

File: README.md [hunk 1]
```diff
@@ -194,3 +194,44 @@ That's it! You should now be ready to cross compile your Raspberry Pi
...
+# Optional: Extending the toolset to support more system dependencies
+
+Let's say your project uses some crate that depends on having openssl
+installed on the system. In this case you have to install the package
+manually into the arm toolset.
+
+Get these packages either from the raspberry, or download them online.
+
+We'll assume that you're running Raspbian (i.e. deb files), but it should be
+straight forward to adapt the steps to your dist/packages. 32 bit is also
```
Comment (Ogeon) [inline README.md]: I think these instructions are the same, regardless if it's a 32 bit or 64 bit system.

File: README.md [hunk 2]
```diff
@@ -194,3 +194,44 @@ That's it! You should now be ready to cross compile your Raspberry Pi
 projects.
 
 Have fun!
+
+
+# Optional: Extending the toolset to support more system dependencies
+
+Let's say your project uses some crate that depends on having openssl
+installed on the system. In this case you have to install the package
+manually into the arm toolset.
```
Comment (Ogeon) [inline README.md]: Could you capitalize ARM?

File: README.md [hunk 3]
```diff
@@ -194,3 +194,44 @@ That's it! You should now be ready to cross compile your Raspberry Pi
...
+assumed from here on.
+
+If you do `apt-cache show libssl1.0.0` on the raspberry, you'll see this in the
+output:
+
+    Filename:    pool/main/o/openssl/libssl1.0.0_1.0.1e-2+rvt+deb7u17_armhf.deb
+
+You should be able to find a match for that under:
+
+http://ftp.debian.org/debian/pool/
```
Comment (Ogeon) [inline README.md]: It would be nice to have an example of a full path, just to make it clear how it may look.

File: README.md [hunk 4]
```diff
@@ -194,3 +194,44 @@ That's it! You should now be ready to cross compile your Raspberry Pi
...
+
+If not, see if it is still on the raspberry under `/var/cache/apt/archive`.
+
+If not, try searching for the filename online.
+
+When you have the dependencies downloaded, extract them into the arm toolchain:
+
+    # cd into the sysroot of the arm toolchain
+    cd /home/vagrant/pi-tools/arm-bcm2708/arm-bcm2708hardfp-linux-gnueabi/arm-bcm2708hardfp-linux-gnueabi/sysroot/
```
Comment (Ogeon) [inline README.md]: Can you replace `/home/vagrant` with `~` to make it more consistent with the rest of the guide?

**Comment by Ogeon** [comment]
I have taken an other look at this, as you can see, and it's very good. I have just commented on a few details that are more about formality or clarity than anything else. I just want it to fit nicely into the rest of the guide, and to make sure that the reader doesn't have to ask any further questions.

I will be happy to merge this if you can revise it and squash the changes into a single commit.

**Comment by tfnico** [comment]
@Ogeon Thanks for the feedback! I'll need some time before I can sit down with this again, but please leave it open.

**Comment by Ogeon** [comment]
No worries, take your time.

**Comment by Ogeon** [comment]
Hi, I just want to check if you are still interested in this pull request.

**Comment by tfnico** [comment]
@Ogeon Sorry, I completely forgot about this one. Pushing it up on my todo-list a bit. Hopefully I can push over the weekend.

**Comment by Ogeon** [comment]
Alright.

**Comment by tfnico** [comment]
@Ogeon OK, I updated the commit. Can you have another look?

**Comment by Ogeon** [comment]
Really nice. Thank you!

**Pull request merged**